src/Css_AtomicTypes.re

```diff
@@ -1535,3 +1535,77 @@ module FontDisplay = {
     | `fallback => "fallback"
     | `optional => "optional";
 };
+
+module Counter = {
+  type style = [ ListStyleType.t | `unset];
+  type t = [ | `counter(string, style) | `counters(string, string, style)];
+
+  let counter = (~style=`unset, name) => `counter((name, style));
+  let counters = (~style=`unset, ~separator="", name) =>
+    `counters((name, separator, style));
+
+  let toString =
+    fun
+    | `counter(counter, style) =>
+      switch (style) {
+      | `unset => "counter(" ++ counter ++ ")"
+      | #ListStyleType.t as t =>
+        "counter(" ++ counter ++ "," ++ ListStyleType.toString(t) ++ ")"
+      }
+    | `counters(counter, string, style) =>
+      switch (style) {
+      | `unset => "counters(" ++ counter ++ ",\"" ++ string ++ "\")"
+      | #ListStyleType.t as t =>
+        "counters("
+        ++ counter
+        ++ ",\""
+        ++ string
+        ++ "\","
+        ++ ListStyleType.toString(t)
+        ++ ")"
+      };
+};
+
+module CounterOperation = {
+  type t = [
+    | `none
+    | `increment(string, int)
```

Comment:
I'm not sure about this API as well. `counterIncrement(CounterOperation.increment(...))` seems redundant. How can I improve it? I'd suggest unifying these three types and methods, but how to name them?
```reasonml
type t = [ | `none | `operation(string, int)];
let make = (~value=0, name) => `operation((name, value));
```
It doesn't seem good.

Reply:
IMO, https://developer.mozilla.org/en-US/docs/Web/CSS/counter-increment, https://developer.mozilla.org/en-US/docs/Web/CSS/counter-reset and https://developer.mozilla.org/en-US/docs/Web/CSS/counter-set are defined as CSS properties. They should have their own atomic types

Reply:
Do you mean three separate modules? Or you say the current solution is all right?

Reply:
Alright, I did it like this, three modules, three variants. It does make sense!